### PR TITLE
Multi version generalization

### PR DESCRIPTION
**Pull request by abbytiz**
@nkubala @cftorres @aaron-prindle 
#2 

(Ideally to be merged before #1 so that pip differ is working accurately before being expanded to more package locations)

### CODE REVIEW

File: .gofmt.sh [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+set -e
+
+files=$(find . -name "*.go" | grep -v iDiff/vendor | grep -v vendor/ | xargs gofmt -l -s)
```
Comment (nkubala) [inline .gofmt.sh]: remove the `grep -v iDiff/vendor`

File: .gofmt.sh [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+if [[ $files ]]; then
+    echo "Gofmt errors in files: $files"
+    exit 1
+fi
+
+files=$(go vet ./structure_tests)
```
Comment (nkubala) [inline .gofmt.sh]: remove this block

File: .gofmt.sh (hunk 1 above)
Comment (abbytiz) [inline .gofmt.sh]: done

File: .gofmt.sh (hunk 2 above)
Comment (abbytiz) [inline .gofmt.sh]: done

File: utils/package_diff_utils.go [hunk 3]
```diff
@@ -48,61 +47,29 @@ type PackageInfo struct {
...
-			continue
-		}
-		if tempPath1[2] == tempPath2[2] {
-			return i, true
-		}
-	}
-	return 0, false
-}
 
 func multiVersionDiff(infoDiff []MultiVersionInfo, key string, map1, map2 map[string]PackageInfo) []MultiVersionInfo {
```
Comment (cftorres) [inline utils/package_diff_utils.go]: maybe more descriptive vars below besides "key" and "value" types would be helpful to understanding what's going on here

Comment (abbytiz) [inline utils/package_diff_utils.go]: I reworked the key/value naming in this function and diffMaps.  Hopefully it's a little clearer what is happening.

**Comment by cftorres** [comment]
lgtm

**Pull request merged**